neo/IO/Helper.cs

```diff
@@ -131,7 +131,14 @@ public static T[] ReadSerializableArray<T>(this BinaryReader reader, int max = 0
 
         public static byte[] ReadVarBytes(this BinaryReader reader, int max = 0x1000000)
         {
-            return reader.ReadBytes((int)reader.ReadVarInt((ulong)max));
+            max = (int)reader.ReadVarInt((ulong)max);
+
+            if (reader.BaseStream.CanSeek)
+            {
+                max = Math.Min((int)(reader.BaseStream.Length - reader.BaseStream.Position), max);
```

Comment:
This can result in the reading of truncated data.

```
0x08123456
```

You should throw an exception instead of returning `0x123456`.

Reply:
Now is returned the data truncated, i do this for prevent the logic change

Reply:
But reading truncated data is not the expected behavior of this function and can cause errors elsewhere.